Sources/App/Models/Bus.swift

```diff
@@ -111,6 +115,56 @@ final class Bus: Hashable, Model {
 	func hash(into hasher: inout Hasher) {
 		hasher.combine(self.id)
 	}
+	// Update the Route that the bus is currently on
+	func updateRoute(selecting routes: [Route]) {		
+		// Calculate the average Coordinate across the stored location points to determine the distance of the 
+		// average direction of the bus to the nearest waypoint along a route.
+		var numCoordinates: Double = 0
+		var longitude: Double = 0
+		var latitude: Double = 0
+		for location in self.locations {
+			longitude += location.coordinate.longitude
+			latitude += location.coordinate.latitude
+			numCoordinates += 1
+		}
+		if (numCoordinates > 0){
+			longitude = longitude / numCoordinates
+			latitude = latitude / numCoordinates
+		}
+		else {
+			self.routeUUID = nil
+		}
+		let avgCoordinate = Coordinate(latitude: latitude, longitude: longitude)
+		var	minDist: Double = 9999
+		var dist: Double = 10000
+		var dist_x1: Double = -1
+		var dist_x2: Double = -1
```

Comment:
You use several [“magic numbers”](https://en.m.wikipedia.org/wiki/Magic_number_(programming)) in this method. It’s generally better to set default/initial values that are more descriptive. Generally, when you need to perform numerical comparison (_e.g._, `<`, `>`, _etc._) on the initial value, opt for the built-in min and max values (_e.g._, [`Int.min`](https://developer.apple.com/documentation/swift/int/min), [`Double.infinity`](https://developer.apple.com/documentation/swift/double/infinity), and others); otherwise, making the variables optional and initializing them to `nil` usually works well.

However, expressively is paramount, so if it wouldn’t be obvious to someone who’s reading the code for the first time what the initial values _mean_, then you might want to employ an enumeration with a payload value. Here’s a simple example:
```swift
enum DistanceValue {
	case base
	case actual(Double)
}
var distanceValue: DistanceValue = .base
let candidates: [Double] = …
for candidate in candidates {
	switch distance {
	case .base:
		distanceValue = .actual(candidate)
	case .actual(let value):
		if candidate < base {
			distanceValue = .actual(candidate)
		}
	}
}
```
This pattern is obviously a lot more verbose, so only fall back on it if you can’t achieve a good level of expressivity with other language features like number-type max/min values or optionals. In the “real world”, the above example should probably just set `distanceValue` to `Double.infinity` and run the less-than comparison on `distanceValue` directly instead of wrapping it in an enumeration. A comment that explains the use of `Double.infinity` would also be a good inclusion.

Let me know if you want additional guidance or help with this; we can chat at the next RCOS meeting.

Reply:
I have some questions regarding optimizations and how to resolve optional types using Swift, but I'll bring this up in the next RCOS meeting.

Reply:
This has been resolved as per our conversation during RCOS.